src/core_editor/editor.rs

```diff
@@ -1,16 +1,38 @@
+use crate::core_editor::get_default_clipboard;
+
 use super::{Clipboard, LineBuffer};
 
 pub struct Editor {
     // Hack: Fix this once coupling in the code is taken care of
     pub line_buffer: LineBuffer,
     cut_buffer: Box<dyn Clipboard>,
+
+    edits: Vec<LineBuffer>,
+    index_undo: usize,
+}
+
+impl Default for Editor {
+    fn default() -> Self {
+        Editor {
+            line_buffer: LineBuffer::new(),
+            cut_buffer: Box::new(get_default_clipboard()),
+
+            // Note: Using list-zipper we can reduce these to one fild
+            edits: vec![],
+            index_undo: 2,
```

Comment:
I wasn't too sure why we are using 2 as an index here?

Reply:
@sherubthakur Because of this line:

`self.lines = self.old_lines.get(self.get_index_undo())?.clone();`

The get requires 1 more and since when we initialize everything we insert a blank string as initial state for undo the index will result initialized as 2